jar-infer/src/test/java/com/uber/nullaway/jarinfer/JarInferTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2018. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.nullaway.jarinfer;
+
+import com.google.common.collect.ObjectArrays;
+import com.google.common.collect.Sets;
+import com.sun.tools.javac.main.Main;
+import com.sun.tools.javac.main.Main.Result;
+import java.util.*;
+import java.util.Arrays;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link com.uber.nullaway.jarinfer}. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("CheckTestExtendsBaseClass")
+public class JarInferTest {
+  private static final String DEFAULT_PACKAGE = "test";
+  private static final String DEFAULT_CLASS = "Test";
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilerUtil compilerUtil;
+
+  @Before
+  public void setup() {
+    compilerUtil = new CompilerUtil(getClass());
+    compilerUtil.setArgs(Arrays.asList("-d", temporaryFolder.getRoot().getAbsolutePath()));
+  }
+
+  private void testTemplate(
+      String testName,
+      String pkg,
+      String cls,
+      String method,
+      String methodSignature,
+      Set<String> expected,
+      String... lines) {
+    Result compileResult =
+        compilerUtil
+            .addSourceLines(cls + ".java", ObjectArrays.concat("package " + pkg + ";\n", lines))
+            .run();
+    Assert.assertEquals(
+        testName + ": test compilation failed!\n" + compilerUtil.getOutput(),
+        Main.Result.OK,
+        compileResult);
+    DefinitelyDerefedParamsDriver defDerefParamDriver = new DefinitelyDerefedParamsDriver();
+    try {
+      Assert.assertTrue(
+          testName + ": test failed!",
+          defDerefParamDriver.verify(
+              defDerefParamDriver.run(
+                  temporaryFolder.getRoot().getAbsolutePath(),
+                  "L" + pkg + "/" + cls,
```

Comment:
... but here is in slash syntax.

So, anything with more than one level deep of package nesting will fail here. Maybe you want to do `pkg. replaceAll(".","/")` here?